### PR TITLE
Implementation of nodal source terms

### PR DESCRIPTION
**Pull request by TomFischer**
Nodal source terms are setup in the project file in the `process_variable` section.
```
<source_terms>
    <source_term>
        <geometrical_set>geometry</geometrical_set>
        <geometry>inner</geometry>
        <type>Nodal</type>
        <value>1</value>
    </source_term>
</source_terms>
```

### CODE REVIEW

File: Documentation/ProjectFile/prj/process_variables/process_variable/source_terms/source_term/t_geometrical_set.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+The name of the geometrical set (see \ref ogs_file_param__gml__name) in which
+the \ref
+ogs_file_param__prj__process_variables__process_variable__source_terms__ource_term_geometry
```
Comment (endJunction) [inline Documentation/ProjectFile/prj/process_variables/process_variable/source_terms/source_term/t_geometrical_set.md]: `ource_term_geometry` -> `source_term_geometry` I guess...

File: MeshGeoToolsLib/CreateSearchLength.h [hunk 2]
```diff
@@ -20,6 +20,7 @@
 
 namespace MeshGeoToolsLib
 {
+inline
 std::unique_ptr<MeshGeoToolsLib::SearchLength> createSearchLengthAlgorithm(
     BaseLib::ConfigTree const& external_config, MeshLib::Mesh const& mesh)
 {
```
Comment (endJunction) [inline MeshGeoToolsLib/CreateSearchLength.h]: Would you move it to cpp, please.

File: ProcessLib/SourceTerms/Tests.cmake [hunk 3]
```diff
@@ -0,0 +1,98 @@
+AddTest(
```
Comment (endJunction) [inline ProcessLib/SourceTerms/Tests.cmake]: I'd move the Tests.cmake content to the corresponding process' Tests.cmake....

File: ProcessLib/SourceTerms/CreateNodalSourceTerm.cpp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    BaseLib::ConfigTree const& config,
+    const NumLib::LocalToGlobalIndexMap& dof_table, std::size_t const mesh_id,
+    std::size_t const node_id, const int variable_id, const int component_id)
+{
+    DBUG("Constructing NodalSourceTerm from config.");
+    //! \ogs_file_param{prj__process_variables__process_variable__source_terms__source_term__type}
+    config.checkConfigParameter("type", "Nodal");
+
+    //! \ogs_file_param{prj__process_variables__process_variable__source_terms__source_term__value}
+    auto const nodal_value = config.getConfigParameter<double>("value");
```
Comment (endJunction) [inline ProcessLib/SourceTerms/CreateNodalSourceTerm.cpp]: The `<value>` tag belongs to the "Nodal" type, so the `ogs_file_param` must be `..._source_term__Nodal__value`. And then you need a new folder `Nodal/` in the `Documentation/.../source_term/`. Compare to [DirichletBC.cpp](https://github.com/ufz/ogs/blob/92f2bf259feabdebdf666d0b1bab0e79620e611e/ProcessLib/BoundaryCondition/DirichletBoundaryCondition.cpp#L59-L63)

**Comment by endJunction** [comment]
I think you didn't use git lfs properly. Please enable the [pre-commit hook](https://github.com/ufz/ogs/blob/master/.hooks/pre-commit)...

**Comment by endJunction** [comment]
I think you didn't use 'git lfs' properly. Please enable the [pre-commit hook](https://github.com/ufz/ogs/blob/master/.hooks/pre-commit).

**Comment by TomFischer** [comment]
If there are no objections I'd like to rebase and merge the branch.

**Comment by wenqing** [comment]
:fast_forward:  if you do not wait jenkins test works

File: ProcessLib/SourceTerms/NodalSourceTerm.cpp [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+void NodalSourceTerm::integrateNodalSourceTerm(
+    const double t,
+    GlobalVector& b) const
+{
+    DBUG("Assemble NodalSourceTerm.");
+
+    MeshLib::Location const l{_mesh_id, MeshLib::MeshItemType::Node, _node_id};
+    auto const index =
+        _dof_table.getGlobalIndex(l, _variable_id, _component_id);
+    b.add(index, _value);
```
Comment (chleh) [inline ProcessLib/SourceTerms/NodalSourceTerm.cpp]: Does that mean that such source terms only work for single nodes? Is an extension to line sources/volumetric sources planned?

File: ProcessLib/SourceTerms/SourceTermBuilder.cpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        });
+    ids.erase(ids_new_end_iterator, std::end(ids));
+
+    DBUG(
+        "Found %d nodes for nodal source term for the variable %d and "
+        "component %d",
+        ids.size(), variable_id, *config.component_id);
+
+    return ProcessLib::createNodalSourceTerm(
+        config.config, dof_table, mesh.getID(), ids[0], variable_id,
```
Comment (chleh) [inline ProcessLib/SourceTerms/SourceTermBuilder.cpp]: IMHO it's very bad to silently drop n-1 of n sound node ids.
More complying with our current standards would be to OGS_FATAL if `ids.size() > 1`.

File: web/content/docs/benchmarks/elliptic/groundwater-flow-dirichlet-nodal-source-term.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+}$$
+The solution of this problem is
+$$
+h(x,y) = \int \int f(\xi, \eta) G(x, y) d \xi d \eta,
+$$
+where $G(x, y)$ is the Green's function. For the example at hand $G(x, y)$ is:
+$$
+G(x, y) = \frac{1}{2 \pi} \ln \sqrt{(x-\xi)^2 + (y-\eta)^2}.
+$$
+With a nodal source term at $(0.0, 0.0)$ the analytical solution is
```
Comment (chleh) [inline web/content/docs/benchmarks/elliptic/groundwater-flow-dirichlet-nodal-source-term.md]: How strong is the source? Unity?

File: ProcessLib/SourceTerms/NodalSourceTerm.cpp [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+      _node_id(node_id),
+      _variable_id(variable_id),
+      _component_id(component_id),
+      _value(value)
+{
+    DBUG("Create NodalSourceTerm.");
+}
+
+void NodalSourceTerm::integrateNodalSourceTerm(
+    const double t,
```
Comment (wenqing) [inline ProcessLib/SourceTerms/NodalSourceTerm.cpp]: @TomFischer  `t` is not used. @TomFischer Would you please send a PR to remove this compilation warning?

**Pull request merged**